test/src/tests-event-logging.js

```diff
@@ -1044,4 +1048,122 @@ describe('event logging', function() {
     });
-});
\ No newline at end of file
+});
+
+describe('event logging without persistence', function() {
+    beforeEach(function() {
+        mParticle._resetForTests(MPConfig);
+        fetchMock.post(urls.events, 200);
+        delete mParticle._instances['default_instance'];
+        mockServer = sinon.createFakeServer();
+        mockServer.respondImmediately = true;
+
+        mockServer.respondWith(urls.identify, [
+            200,
+            {},
+            JSON.stringify({ mpid: testMPID, is_logged_in: false }),
+        ]);
+
+        mParticle.init(apiKey, {
+            ...window.mParticle.config,
+            usePersistence: false,
+        });
+    });
+
+    afterEach(function() {
+        mockServer.restore();
+        fetchMock.restore();
+        mParticle._resetForTests(MPConfig);
+    });
+
+    it('should log an event', function(done) {
+        window.mParticle.logEvent(
+            'Test Event',
+            mParticle.EventType.Navigation,
+            { mykey: 'myvalue' },
+            { myFlags: 'some-flag' }
+        );
+        const testEvent = findEventFromRequest(fetchMock.calls(), 'Test Event');
+        const testEventBatch = findBatch(fetchMock.calls(), 'Test Event');
+
+        expect(testEvent.data.event_name, 'event name').to.equal('Test Event');
+        expect(
+            testEvent.data.session_start_unixtime_ms,
+            'event session start time'
+        ).to.not.equal(null);
+        expect(testEvent.data.session_uuid, 'event session uuid').to.not.equal(
+            null
+        );
+        expect(
+            testEvent.data.source_message_id,
+            'event source message id'
+        ).to.not.equal(null);
+        expect(
+            testEvent.data.timestamp_unixtime_ms,
+            'event timestamp'
+        ).to.not.equal(null);
+
+        expect(
+            testEvent.data.custom_event_type,
+            'event custom event type'
+        ).to.equal('navigation');
+
+        expect(
+            testEvent.data.custom_attributes,
+            'event custom attributes'
+        ).to.eqls({ mykey: 'myvalue' });
+
+        expect(testEvent.data.custom_flags, 'event custom flags').to.eqls({
+            myFlags: 'some-flag',
+        });
+
+        expect(testEventBatch.mpid, 'batch mpid').to.equal(testMPID);
+        expect(testEventBatch.consent_state, 'batch consent state').to.eqls(
+            null
+        );
+        expect(testEventBatch.mp_deviceid, 'batch device id').to.not.equal(
+            null
+        );
+        expect(testEventBatch.sdk_version, 'batch sdk version').to.not.equal(
+            null
+        );
+        expect(
+            testEventBatch.source_request_id,
+            'batch source_request_id'
+        ).to.not.equal(null);
+        expect(
+            testEventBatch.timestamp_unixtime_ms,
+            'batch timestamp'
+        ).to.not.equal(null);
+        expect(testEventBatch.user_attributes, 'batch user attributes').to.eqls(
```

Comment:
i just did a search for what `eqls` means.  apparently it's deep equality. Is there a difference than between `to.eqls` and `to.deep.equal`?   I prefer the latter since it's more intuitive